File: typescript/src/line.ts
import { newPolarPoint, PolarPoint } from "./points/polarPoint";

function newLine (point1: Point, point2: Point) {
    return {    
        draw() {
            const x1: number = point1.getX();
            const y1: number = point1.getY();
            const x2: number = point2.getX();
            const y2: number = point2.getY();
            
            // assume implementation 
    
            console.log("Line drawn from (" + x1 + ", " + y1 +
                        ") to (" + x2 + ", " + y2 + ")");
        }
    }
}

export { newLine }

File: java/src/main/java/edu/cmu/cs/cs214/rec02/Line.java
package edu.cmu.cs.cs214.rec02;

import edu.cmu.cs.cs214.rec02.points.*;

public class Line {
    private Point point1;
    private Point point2;

    Line(Point point1, Point point2) {
        this.point1 = point1;
        this.point2 = point2;
    }

    void draw() {
        int x1 = point1.getX();
        int y1 = point1.getY();
        int x2 = point2.getX();
        int y2 = point2.getY();
        
        // assume implementation 

        System.out.println("Line drawn from (" + x1 + ", " + y1 +
                            ") to (" + x2 + ", " + y2 + ")");
    }
}
